cumulus_library/databases.py

```diff
@@ -0,0 +1,213 @@
+"""Abstraction layers for supported database backends (e.g. AWS & DuckDB)"""
+
+import abc
+import datetime
+import json
+import os
+from pathlib import Path
+from typing import Protocol
+
+import cumulus_fhir_support
+import duckdb
+import pandas
+import pyarrow
+import pyathena
+from pyathena.common import BaseCursor as AthenaCursor
+from pyathena.pandas.cursor import PandasCursor as AthenaPandasCursor
+
+
+# TODO: find a standard Protocol to use instead of homebrew
```

Comment:
some cursory googling indicates there might not be one

Reply:
:+1: removed this comment